eventsourcing/src/main/java/org/axonframework/eventsourcing/eventstore/BatchingEventStorageEngine.java

```diff
@@ -247,9 +250,19 @@ public boolean tryAdvance(Consumer<? super T> action) {
                 if (iterator != null && batchSize > sizeOfLastBatch && !fetchUntilEmpty) {
                     return false;
                 }
+                if (contiguousSmallResultSeen) {
+                    return false;
+                }
                 List<? extends T> items = fetchFunction.apply(lastItem);
+                int numItems = items.size();
+                if (numItems > 0 && numItems < smallResultTreshold) {
+                    DomainEventData<T> firstFetchedItem = (DomainEventData<T>)items.get(0);
```

Comment:
From the spacing of brackets and operations, I am guessing you're not using the current style guide for this change.
Would you mind setting it for this PR? You can find it [here](https://github.com/AxonFramework/AxonFramework/blob/master/axon_code_style.xml) by the way.